diofant/integrals/tests/test_meijerint.py

```diff
@@ -651,3 +651,8 @@ def test_issue_6860():
 def test_issue_8368():
     assert meijerint_indefinite(cosh(x)*exp(-x*t), x) == (
         (-t - 1)*exp(x) + (-t + 1)*exp(-x))*exp(-t*x)/2/(t**2 - 1)
+
+
+def test_issue_4311():
+    assert meijerint_indefinite(x*abs(9 - x**2), x) == \
+        -81*meijerg(((3, 1), (2,)), ((1,), (2, 0)), x**2/9)
```

Comment:
I'm not sure if this is a correct answer.

Diofant can't expand this yet, but Mathematica can:
```
In[1]:= -81*MeijerG[{{3, 1}, {2}}, {{1}, {2, 0}}, (1/9)*x^2]

Out[1]= 81 MeijerG[{{3}, {}}, {{}, {0}}, x^2/9]
```

Note that:
```
In[8]:= Out[1] /. x -> 2.0

Out[8]= 0.

In[9]:= Out[1] /. x -> 4.0

Out[9]= 24.5
```

In turn, Diofant can hyperexpand Out[1]:
```
In [16]: integrate(x*abs(9 - x**2), x)
Out[16]: 
            ⎛           │  2⎞
    ╭─╮1, 2 ⎜3, 1   2   │ x ⎟
-81⋅│╶┐     ⎜           │ ──⎟
    ╰─╯3, 3 ⎝ 1    2, 0 │ 9 ⎠

In [17]: 81*meijerg(((3,), ()), ((), (0,)), x**2/9)
Out[17]: 
           ⎛     │  2⎞
   ╭─╮0, 1 ⎜3    │ x ⎟
81⋅│╶┐     ⎜     │ ──⎟
   ╰─╯1, 1 ⎝   0 │ 9 ⎠

In [18]: hyperexpand(_)
Out[18]: 
   ⎛⎧                             │ 2│     ⎞
   ⎜⎪                             │x │     ⎟
   ⎜⎪          0              for ──── < 1 ⎟
   ⎜⎪                              9       ⎟
   ⎜⎪                                      ⎟
   ⎜⎪ 4 ⎛ 1     1      1  ⎞        │1 │    ⎟
   ⎜⎪x ⋅⎜─── - ──── + ────⎟  for 9⋅│──│ < 1⎟
81⋅⎜⎨   ⎜162      2      4⎟        │ 2│    ⎟
   ⎜⎪   ⎝      9⋅x    2⋅x ⎠        │x │    ⎟
   ⎜⎪                                      ⎟
   ⎜⎪         ⎛     │  2⎞                  ⎟
   ⎜⎪ ╭─╮0, 1 ⎜3    │ x ⎟                  ⎟
   ⎜⎪ │╶┐     ⎜     │ ──⎟      otherwise   ⎟
   ⎜⎪ ╰─╯1, 1 ⎝   0 │ 9 ⎠                  ⎟
   ⎝⎩                                      ⎠
In [19]: simplify(piecewise_fold(_))
Out[19]: 
⎧                             │ 2│     
⎪                             │x │     
⎪          0              for ──── < 1 
⎪                              9       
⎪                                      
⎪     4                                
⎪    x       2   81            │1 │    
⎨    ── - 9⋅x  + ──      for 9⋅│──│ < 1
⎪    2           2             │ 2│    
⎪                              │x │    
⎪                                      
⎪           ⎛     │  2⎞                
⎪   ╭─╮0, 1 ⎜3    │ x ⎟                
⎪81⋅│╶┐     ⎜     │ ──⎟    otherwise   
⎩   ╰─╯1, 1 ⎝   0 │ 9 ⎠                

In [20]: simplify(_.diff(x))
Out[20]: 
⎧                                                        │ 2│     
⎪                                                        │x │     
⎪                        0                           for ──── < 1 
⎪                                                         9       
⎪                                                                 
⎪                      ⎛ 2    ⎞                           │1 │    
⎪                  2⋅x⋅⎝x  - 9⎠                     for 9⋅│──│ < 1
⎪                                                         │ 2│    
⎨                                                         │x │    
⎪                                                                 
⎪    ⎛        ⎛     │  2⎞             ⎛     │  2⎞⎞                
⎪    ⎜╭─╮0, 1 ⎜2    │ x ⎟     ╭─╮0, 1 ⎜3    │ x ⎟⎟                
⎪162⋅⎜│╶┐     ⎜     │ ──⎟ + 2⋅│╶┐     ⎜     │ ──⎟⎟                
⎪    ⎝╰─╯1, 1 ⎝   0 │ 9 ⎠     ╰─╯1, 1 ⎝   0 │ 9 ⎠⎠                
⎪─────────────────────────────────────────────────    otherwise   
⎪                        x                                        
⎩                                                                 
```
Case ``Abs(x)<3`` is wrong and case ``Abs(x)>3`` is twice as wrong.

The Diofant's ``hyperexpand`` seems to be correct here, see e.g. numerical tests above.

Either meijerint_indefinite is wrong, or Mathematica's simplification rules for meijerg.  Second case may be possible, e.g:
```
In [25]: Out[16].evalf(subs={x: 2.0})
Out[25]: -14.6666666666667

In [26]: Out[16].evalf(subs={x: 4.0})
Out[26]: -66.1666666666667

In [27]: Out[17].evalf(subs={x: 2.0})
Out[27]: 0

In [28]: Out[17].evalf(subs={x: 4.0})
Out[28]: 24.5000000000000
```